File: submissions/sm_012_gaurav/week_03/day_6/assignments/stringsPlay.js
//Function to find index of first occurance of a Character in a string.
function indexOf(input, character) {
    var index = -1;
    for (var i = 0; i < input.length; i++) {
        if (input[i] == character) {
            index = i;
            break;
        }
    }
    return index;
}


//Function to convert string into lower case.
function lowerCase(input) {
    capitalLetters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
    smallLetters = 'abcdefghijklmnopqrstuvwxyz';
    var lowerCaseString = '';
    for (var i = 0; i < input.length; i++) {
        var indexInCapital = indexOf(capitalLetters, input[i]);
        if (indexInCapital != -1) {
            lowerCaseString += smallLetters[indexInCapital];
        }
        else {
            lowerCaseString += input[i];
        }
    }
    return lowerCaseString;
}

//Function to split a string on given character.
function splitString(input, character) {
    var arr = [];
    var str = '';
    for (var i = 0; i < input.length; i++) {
        if (input[i] != character) {
            str += input[i];
        }
        else {
            if (str.length != 0)
                arr.push(str)
            str = '';
        }
    }
    if (str.length != 0)
        arr.push(str);
    return arr;
}

//Function to capitalize the string.
function capitalizeString(input) {
    capitalLetters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
    smallLetters = 'abcdefghijklmnopqrstuvwxyz';
    var capitalizedString = '';
    if (indexOf(smallLetters, input[0]) != -1) {
        capitalizedString += capitalLetters[indexOf(smallLetters, input[0])];
    }
    else {
        capitalizedString += input[0];
    }
    for (var i = 1; i < input.length; i++) {
        var indexInCapital = indexOf(capitalLetters, input[i]);
        if (indexInCapital != -1) {
            capitalizedString += smallLetters[indexInCapital];
        }
        else {
            capitalizedString += input[i];
        }
    }
    return capitalizedString;
}


//Function to join array of string through a character and return a string.
function joinArray(input, character) {
    var str = '';
    for (var i = 0; i < input.length; i++) {
        if (i != input.length-1) {
            str += (input[i]+character);
        }
        else {
            str += input[i];
        }
    }
    return str;
}

//Function to convert string into camelcase.
function camelCase(input) {
    input = splitString(input, ' ');
    for (var i = 0; i < input.length; i++) {
        if (i == 0) {
            input[i] = lowerCase(input[i]);
        }
        else {
            input[i] = capitalizeString(input[i]);
        }
    }
    return joinArray(input, '');
}

//Function to convert string into snakecase.
function snakeCase(input) {
    input = splitString(input, ' ');
    for (var i = 0; i < input.length; i++) {
        input[i] = lowerCase(input[i]);
    }
    return joinArray(input, '_');
}

//Function to convert string into hyphencase.
function hyphenCase(input) {
    input = splitString(input, ' ');
    for (var i = 0; i < input.length; i++) {
        input[i] = lowerCase(input[i]);
    }
    return joinArray(input, '-');
}

console.log(camelCase('Hello Masai SCHOOL'));    //helloMasaiSchool
console.log(snakeCase('Hello Masai SCHOOL'));    //hello_masai_school
console.log(hyphenCase('Hello Masai SCHOOL'));   //hello-masai-school

File: submissions/sm_012_gaurav/week_13/day_5/session_1/strings_play.py
def splitStr(s, splitChar):
    l = []
    st = ''
    for a in s:
        if a == splitChar:
            if len(st) != 0:
                l.append(st)
                st = ''
        else:
            st += a
    if len(st) != 0:
        l.append(st)
    return l

def indexOf(s, c):
    idx = 0
    for i,a in enumerate(s):
        if a == c:
            idx = i
            break
    return idx

def capitalizeStr(s):
    st = ''
    capitalLetters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
    smallLetters = 'abcdefghijklmnopqrstuvwxyz'
    for i,a in enumerate(s):
        if i == 0:
            if a not in capitalLetters:
                st += capitalLetters[indexOf(smallLetters, a)]
            else:
                st += a
        else:
            if a in capitalLetters:
                st += smallLetters[indexOf(capitalLetters, a)]
            else:
                st += a
    return st

def lowerCaseStr(s):
    st = ''
    capitalLetters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
    smallLetters = 'abcdefghijklmnopqrstuvwxyz'
    for a in s:
        if a in capitalLetters:
            st += smallLetters[indexOf(capitalLetters, a)]
        else:
            st += a
    return st

def camel_case(s, splitChar = ' '):
    l = splitStr(s, splitChar)
    changedStr = ''
    for i in range(len(l)):
        if i == 0:
            changedStr += lowerCaseStr(l[i])
        else:
            changedStr += capitalizeStr(l[i])
    return changedStr

def hyphen_case(s, splitChar = ' '):
    l = splitStr(s, splitChar)
    changedStr = ''
    for i in range(len(l)):
        if i != len(l) - 1:
            changedStr += lowerCaseStr(l[i])+'-'
        else:
            changedStr += lowerCaseStr(l[i])
    return changedStr

def snake_case(s, splitChar = ' '):
    l = splitStr(s, splitChar)
    changedStr = ''
    for i in range(len(l)):
        if i != len(l) - 1:
            changedStr += lowerCaseStr(l[i])+'_'
        else:
            changedStr += lowerCaseStr(l[i])
    return changedStr
